text/nodejs/0013-install-bom.md

```diff
@@ -61,36 +61,70 @@ the following fields for each module:
 All of these are fields that we aim to support per [the Paketo BOM
 RFC](https://github.com/sophiewigmore/rfcs/blob/8b1e8c9ed6201313f47c5897223cbffb265e96ed/text/0028-bill-of-materials.md)
 
-### Buildpack Additions
+In order to support this tool in both the online and offline cases, we will
+pre-compile the tool with it's dependencies, and provide it as a dependency
+hosted on the [dep-server](https://github.com/paketo-buildpacks/dep-server).
+
+### Language Family Additions
 
 Using the CycloneDX tool to generate the BOM will involve some changes to the
-NPM Install and Yarn Install Buildpacks.
+Node.JS language family buildpack order groups.
+
+* A single BOM generation buildpack will be added. It's primary functions are:
+  * perform the BOM dependency tool retrieval from the dep-server,
+  * execute BOM generation with the tool (`cyclonedx-bom -o bom.json`), and
+  * contribute the BOM to the layer metadata.
+
+* For each of the three Node.JS buildpack order groups, the BOM generation
+  buildpack will run directly before the start command buildpack. This means:
+  * directly before the `yarn-start` buildpack in the Yarn order group,
+  * directly before the `npm-start` buildpack in the NPM order group, and
+  * directly before the `node-start` buildpack in the no-package-manager order group.
+
+Users of the Node.JS buildpack can opt out of this behaviour by creating a
+custom order grouping that does not include the BOM Generation buildpack, or
+they can set a `ENABLE_MODULE_BOM` flag during container build-time to `false`.
+The default for this flag will be `true`.
+
+#### Detection
+
+The buildpack will pass detection in two different cases:
+
+* Detection will pass if there is a `node_modules` directory in the source directory.
+  * Requires `node`
+  * Provides none
 
-* After the `npm install` or `yarn install` command is run during the build
-  phase, we should run `npm install -g @cyclonedx/bom` to install the tool on
-  the container.
+* Detection will also pass if there is no `node_modules` directory
+  * Requires {`node`, `node_modules`}
+  * Provides none
 
-* Run `cyclonedx-bom -o bom.json` from the working directory where the
-  `node_modules` are located
+#### Build
 
-* Implement some code to unmarshal the JSON Bill of Materials and transform
-  each BOM entry into a
-  [`postal.Dependency`](https://github.com/paketo-buildpacks/packit/blob/c5a40518f2c6bd913ade999b9e2d58d6892d2ea9/postal/buildpack.go#L12)
-  type.
+The build phase will perform a few tasks as mentioned above.
 
-* (Optionally) generate `CPE` and `pURL` from the data and add it to the
+1. Perform the BOM dependency tool retrieval from the dep-server. It will get
+   added to a build-time layer. It will contribute a BOM entry for the tool itself
```

Comment:
>  It will get added to a build-time layer

Why is this tool added to a layer (& a BOM entry for the tool itself added) instead of just using the tool to generate the BOM and throwing away the tool?

Reply:
We wanted to cache the tool so that if any node modules change, we can still generate the BOM again without having to re-download the tool.

A BOM entry for the tool is necessary because it's something we are installing onto the image, and we need to record of everything we install via our buildpacks, same as we do for node engine and. yarn